src/carousel/carousel.js

```diff
@@ -421,11 +420,24 @@ angular.module('ui.bootstrap.carousel')
 
 .value('$carouselSuppressWarning', false)
 
+.controller('CarouselController', ['$scope', '$element', '$controller', '$log', '$carouselSuppressWarning', function($scope, $element, $controller, $log, $carouselSuppressWarning) {
+  if (!$carouselSuppressWarning) {
+    $log.warn('CarouselController is now deprecated. Use UibCarouselController instead.');
+  }
+
+  angular.extend(this, $controller('UibCarouselController', {
+    $scope: $scope,
+    $element: $element
+  }));
+}])
+
 .directive('carousel', ['$log', '$carouselSuppressWarning', function($log, $carouselSuppressWarning) {
   return {
-    restrict: 'EA',
     transclude: true,
     replace: true,
+    controller: 'CarouselController',
+    controllerAs: 'carousel',
```

Comment:
Excuse me Chris, but after a couple of months away of the project, there is lots of new features done and I am afraid that you're barking to the wrong tree. I totally understand that.

Imagine we make a mistake a we push a commit where we get a directive and we remove it entirely. Then we want to create another PR to put back the content we deleted. We are simply doing a chore to put it back like it was before. We are not creating multiple PRs with all the features we deleted one by one. We are just restoring something that was there in a first place.